### PR TITLE
Update backfill job, and setup a cronjob in prod

### PR DESCRIPTION
**Pull request by severo**

### CODE REVIEW

File: jobs/cache_maintenance/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+- `metrics`: compute and store the cache and queue metrics
+- `skip`: do nothing
+
+## Configuration
+
+The script can be configured using environment variables. They are grouped by scope.
+
+### Actions
+
+Set environment variables to configure the job (`DATABASE_MIGRATIONS_` prefix):
```
Comment (AndreaFrancis) [inline jobs/cache_maintenance/README.md:17]:
```suggestion
Set environment variables to configure the job (`CACHE_MAINTENANCE_` prefix):
```

File: jobs/cache_maintenance/src/cache_maintenance/backfill.py [hunk 2]
```diff
@@ -4,33 +4,45 @@
...
+        dataset = dataset_info.id
+        if not dataset:
+            # should not occur
+            continue
+        dataset_state = DatasetState(dataset=dataset, processing_graph=processing_graph, revision=dataset_info.sha)
+        if dataset_state.should_be_backfilled:
+            backfilled_datasets += 1
+            created_jobs += dataset_state.backfill()
+
+        analyzed_datasets += 1
```
Comment (AndreaFrancis) [inline jobs/cache_maintenance/src/cache_maintenance/backfill.py:43]: Is it needed twice? (Line 32)

**Review by AndreaFrancis — APPROVED** [review]
backfill/upgrade Looks better now, thanks!

File: jobs/cache_maintenance/src/cache_maintenance/backfill.py (hunk 2 above)
Comment (severo) [reply jobs/cache_maintenance/src/cache_maintenance/backfill.py]: well seen, thanks

https://github.com/huggingface/datasets-server/pull/1077/commits/bcb42402d20b02a35d73f2c839bf69f2e1251a6a

**Review by AndreaFrancis — APPROVED** [review]

**Pull request merged**